bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/model/HmDatapoint.java

```diff
@@ -150,19 +150,34 @@ public int getOptionIndex(String option) {
      */
     public String getOptionValue() {
         if (options != null && value != null) {
-            int idx = 0;
-            if (value instanceof Integer) {
-                idx = (int) value;
-            } else {
-                idx = Integer.parseInt(value.toString());
-            }
+            int idx = getIntegerValue();
```

Comment:
As getIntegerValue can return null, there is probably a test to include here before casting it to an int.

Reply:
Added null check